repositories/goal.repository.js

```diff
@@ -0,0 +1,14 @@
+const MongoStorage = require("../db/mongo.storage");
+
+module.exports = new (class GoalRepository extends MongoStorage {
+    constructor() {
+        super("goal");
+    }
+
+    getGoalSuccessCountById(goalId) {
+        return this.retrieve(goalId)?.success_count ?? null;
```

Comment:
Could you explain the use of ?? operator? I know that it checks of the left operand is nullis, and if so, returns the right operand. If not, returns the left operand. So, if success count is null by itself, you could just return it.

Reply:
It is also used for undefined in this case, making sure that return value is either a real value or null but not undefined, as we used as a convention in other parts of the code

Reply:
thank for clarifying, great job